src/test/java/net/minecraftforge/test/CustomEntitySelectorTest.java

```diff
@@ -0,0 +1,67 @@
+package net.minecraftforge.test;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.Lists;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.util.text.TextComponentString;
+import net.minecraftforge.fml.common.IEntitySelectorFactory;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.common.event.FMLInitializationEvent;
+import net.minecraftforge.fml.common.registry.GameRegistry;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+@Mod(modid = "customentityselectortest", name = "Custom Entity Selector Test", version = "1.0", acceptableRemoteVersions = "*")
```

Comment:
Please add a short javadoc with steps to use in-game to confirm that everything is working with this test.
Someone should be able to use this test to confirm everything is working, without reading the code. This also ensures that any bugs in the code can be checked against this javadoc specification if necessary.